src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,42 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Machine;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        MachineProducer<? extends Machine> machineProducer;
+        switch (type.getSimpleName()) {
+            case "Bulldozer":
+                machineProducer = new BuildozerProducer();
+                return new ArrayList<>(machineProducer.get());
+            case "Excavator":
+                machineProducer = new ExcavatorProducer();
+                return new ArrayList<>(machineProducer.get());
+            case "Truck":
+                machineProducer = new TruckProducer();
+                return new ArrayList<>(machineProducer.get());
+            default:
+                return new ArrayList<>();
```

Comment:
I'd like to leave here several points:
- you are creating new producers each time the method is called.
- I'd recommend creating a map as a class field (using class as a key and producer as a value) and using this map instead of switch
- let's use Collections.emptyList() in case we are going to return empty list